src/TSTransformer/util/wrapReturnIfLuaTuple.ts

```diff
@@ -45,7 +45,7 @@ function shouldWrapLuaTuple(node: ts.CallExpression, exp: luau.Expression) {
 }
 
 export function wrapReturnIfLuaTuple(state: TransformState, node: ts.CallExpression, exp: luau.Expression) {
-	if (isLuaTupleType(state)(state.getType(node)) && shouldWrapLuaTuple(node, exp)) {
+	if (shouldWrapLuaTuple(node, exp) && isDefinitelyType(state, state.getType(node), node, isLuaTupleType(state))) {
```

Comment:
Why did you flip these conditions? Checking `isLuaTupleType()` should be relatively cheap compared to `shouldWrapLuaTuple()`, so anything that isn't a `LuaTuple<T>` will early out.

Reply:
I would doubt that. A LuaTuple check is a full `type.getProperty` which will be slow. The should wrap check is only checkig the node kinds which should be super fast.

The check was reversed because it allows a more lenient type, if we aren't going to wrap it anyways. Because then we don't need to know if it's LuaTuple.